### DISCUSSION TITLE
Add property on NodeSet to specify an alternative baseClassName to use when there is only one child node in the set

### DISCUSSION THREAD

**Discussion by StephenR1** [caplin/FlexLayout#416] [Ideas]
When there is only one tab in a tabset (i.e. for simple layouts), It would be much nicer for the details of the tab window to appear more like a 'Title Bar' than a tab. (E.g. centred text and no 'tab frame')

I think this could be fairly easily implemented by defining a class as part of the properties of the layout (or the tab set) to be used when the number of children in the tabset === 1. The css could then be different for a class on its own vs a set with multiple children.

Making it a separate property should make it easy to make the change without affecting any existing operation?

**Comment by nealus** [comment]
I've basically implemented that in v0.7.13, see the Mosaic Style layout in the demo

**Reply by StephenR1** [reply]
Thanks very much Nealus.

I think the difference here is that that example does not allow multiple tabs in a tabset.

I was thinking more of a case where multiple tabs were allowed and looked just like now but if there was only one tab in the tabset, it would (optionally) use a different styling class

The idea is that a single tab would look more just like a 'window' but would 'transform' into a tab type look and feel as soon as there was more than one to pick from in the tabset.

I had 'frigged' this before by making the tabbar height to 0 and using a header on the tabset as a conditional thing when there was only one child but that fell foul of not being able to merge a named tabset into another one.

Thanks very much again for the great component though. It is so useful!

Stephen

**Reply by StephenR1** [reply]
I also realise that my solution is slightly more invasive than it would have been because the rendering is in the tab rather than the tabset so it probably either needs a flag sent to say that the tab is the only child in the tabset or doing a search on the model to see who the parent is and then using the info. The first way is more tightly coupled but possibly faster I guess.

**Comment by nealus** [comment]
"The idea is that a single tab would look more just like a 'window' but would 'transform' into a tab type look and feel as soon as there was more than one to pick from in the tabset."

That does happen, in the Mosaic Style demo I've added the attribute tabSetEnableDrop=false that prevents dropping into the tabset, but in the demo you can still press the AddActive button to add one, it will then change into a usual tabset

**Comment by StephenR1** [comment]
Ah yes. I didn't try that! Thank you. That looks just the thing.

Is there a way to have all of the features of that but still allow dropping a tab/tabset onto the window to add the tab(s)?

**Reply by nealus** [reply]
just remove the global attribute tabSetEnableDrop from the mosaic layout ( i added that to make the layout like Mosaic layout, just headed panels that can be resized and moved, but don't become tabs). So for your case just add global attribute tabSetEnableSingleTabStretch to your layout

**Reply by StephenR1** [reply]
Thank you!

(I feel very silly for not having looked a the layout JSON before posting my reply...)

That is brilliant!

**Reply by StephenR1** [reply]
I upgraded and added that flag and it works an absolute treat!

Thank you so much for the super-fast turn around!
